src/mem/PoolingAllocator.h

```diff
@@ -11,6 +11,8 @@
 
 #include "mem/forward.h"
 
+#include <utility>
```

Comment:
This (correct) change looks out of scope, but it is needed in this PR to compile the new test code:

```
../src/mem/PoolingAllocator.h:31:102: error: ‘forward’ is not a member of ‘std’
../src/mem/PoolingAllocator.h:13:1: did you forget to ‘#include <utility>’?
```

I am not sure why it was not needed during earlier stages of this PR development, but I suspect it could be because a different order of `#include` directives hid this problem.